tests/js/navButtonsTests.js

```diff
@@ -0,0 +1,111 @@
+        jqUnit.assertEquals("The model value has been updated", currentPanelNum, that.model.currentPanelNum);
+
+        var msg,
+            start = that.options.panelStartNum,
+            end = that.options.panelTotalNum,
+            backButton = that.locate("back"),
+            nextButton = that.locate("next"),
+            backTooltip = that.backTooltip,
+            nextTooltip = that.nextTooltip;
+
+        if (currentPanelNum === start) {
+            msg = "On the start panel";
+            gpii.tests.verifyStates(msg, backButton, nextButton, {
+                backDisabled: true,
+                backVisible: false,
+                nextDisabled: false,
+                nextVisible: true
+            });
+            gpii.tests.verifyLabels(msg, backButton, backTooltip, nextButton, nextTooltip, {
+                backLabel: undefined,
+                nextLabel: that.options.strings.start
+            });
+        }
+
+        if (currentPanelNum > start && currentPanelNum < end) {
+            msg = "On a panel in btw the start and the last panels";
+            gpii.tests.verifyStates(msg, backButton, nextButton, {
+                backDisabled: false,
+                backVisible: true,
+                nextDisabled: false,
+                nextVisible: true
+            });
+            gpii.tests.verifyLabels(msg, backButton, backTooltip, nextButton, nextTooltip, {
+                backLabel: that.options.strings.back,
+                nextLabel: that.options.strings.next
+            });
+        }
+
+        if (currentPanelNum === end) {
+            msg = "On the last panel";
+            gpii.tests.verifyStates(msg, backButton, nextButton, {
+                backDisabled: false,
+                backVisible: true,
+                nextDisabled: false,
+                nextVisible: true
+            });
+            gpii.tests.verifyLabels(msg, backButton, backTooltip, nextButton, nextTooltip, {
+                backLabel: that.options.strings.back,
+                nextLabel: that.options.strings.finish
+            });
+        }
+    };
+
+    jqUnit.test("Nav buttons", function () {
+        jqUnit.expect(34);
+
+        var that = gpii.firstDiscovery.navButtons(".gpiic-nav", {
+            panelTotalNum: 6
+        });
+
+        var backButton = that.locate("back"),
+            nextButton = that.locate("next");
+
+        that.applier.change("currentPanelNum", 1);
+        jqUnit.assertNotUndefined("The tooltip for the back button has been created", that.backTooltip);
+        jqUnit.assertNotUndefined("The tooltip for the next button has been created", that.nextTooltip);
+        gpii.tests.verifyButtons(that, 1);
+
+        nextButton.click();
+        gpii.tests.verifyButtons(that, 2);
+
+        backButton.click();
+        gpii.tests.verifyButtons(that, 1);
+
+        that.applier.change("currentPanelNum", 6);
+        gpii.tests.verifyButtons(that, 6);
```

Comment:
can you leave comments for all of these to explain what it is that they are testing?